src/shutdown_manager.ts

```diff
@@ -31,9 +35,25 @@ export default class ShutdownManager {
         const key = this.shutDownKey(details.instanceId);
         const res = await this.redisClient.get(key);
         ctx.logger.debug('Read shutdown status', { key, res });
-        if (res == 'shutdown') {
-            return true;
-        }
-        return false;
+        return res == 'shutdown';
+    }
+
+    async setScaleDownProtected(
+        ctx: Context,
+        details: InstanceDetails,
+        protectedTTL: number,
+        mode = 'isScaleDownProtected',
+    ): Promise<boolean> {
+        const key = this.protectedKey(details.instanceId);
+        ctx.logger.debug('Writing protected mode', { key, mode });
+        await this.redisClient.set(key, mode, 'ex', protectedTTL);
+        return true;
+    }
+
+    async isScaleDownProtected(ctx: Context, details: InstanceDetails): Promise<boolean> {
```

Comment:
Same question here: do we need the full instance details here?  Seems like if we're only using instanceId, then passing only that value would work and not require knowing the other instance details?  I'm open to discussing, don't feel strongly about it.